src/rust/ensogl/src/system/gpu/data/texture/class.rs

```diff
@@ -45,19 +45,65 @@ impl Drop for TextureBindGuard {
 
 
 
+// ==================
+// === Parameters ===
+// ==================
+
+/// Parameters that need to be set when binding a texture.
+#[derive(Copy, Clone, Debug)]
+pub struct Parameters{
+    /// Specifies the value for `Context::TEXTURE_MIN_FILTER`.
```

Comment:
This doc is not understandable. In order to understand it you need to know what `Context::TEXTURE_MIN_FILTER`. All docs should explain semantically the meaning of things. It should be described here really short what min filter is used for actually. On the other hand, after you create newtypes and add docs there, documentation of these fields could be removed and we can even use `allow(missing_docs)` as the docs would already be there.